// ngx-glide/esm2020/lib/ngx-glide.module.mjs
import { CommonModule } from '@angular/common';
import { NgModule } from '@angular/core';
import { NgxGlideComponent } from './ngx-glide.component';
import * as i0 from "@angular/core";
export class NgxGlideModule {
}
NgxGlideModule.ɵfac = i0.ɵɵngDeclareFactory({ minVersion: "12.0.0", version: "14.0.1", ngImport: i0, type: NgxGlideModule, deps: [], target: i0.ɵɵFactoryTarget.NgModule });
NgxGlideModule.ɵmod = i0.ɵɵngDeclareNgModule({ minVersion: "14.0.0", version: "14.0.1", ngImport: i0, type: NgxGlideModule, declarations: [NgxGlideComponent], imports: [CommonModule], exports: [NgxGlideComponent] });
NgxGlideModule.ɵinj = i0.ɵɵngDeclareInjector({ minVersion: "12.0.0", version: "14.0.1", ngImport: i0, type: NgxGlideModule, imports: [CommonModule] });
i0.ɵɵngDeclareClassMetadata({ minVersion: "12.0.0", version: "14.0.1", ngImport: i0, type: NgxGlideModule, decorators: [{
            type: NgModule,
            args: [{
                    declarations: [NgxGlideComponent],
                    imports: [CommonModule],
                    exports: [NgxGlideComponent],
                }]
        }] });
